File: examples/IOP-Dynamic-ID-Assignment/CSharpClient_5/src/urn_jts_DiscoveryClient_1_0/InternalEvents/RHP_Timeout.cs
using JTS;
using System;
using System.Collections;

namespace urn_jts_DiscoveryClient_1_0
{

public class RHP_Timeout : InternalEvent
{
	/**
	 * Returns the number of bytes the used data members of the class occupies
	 * in the buffer. This is not the number of bytes the data type occupies in 
	 * Java, but the bytes expected on the wire.
	 * 
	 * @return
	 */
	public override int getSize()
	{
		int size = 0;
		
		
		return size;
	}
	
	public override void encode(byte[] bytes, int pos)
	{
		
		if (bytes == null)
		{
			return;
		}
		
		
	}
	
	public override void decode(byte[] bytes, int pos)
	{
		
		if (bytes == null)
		{
			return;
		}
		
		if(pos < bytes.Length)
		{
			for(int i = pos; i<bytes.Length; i++)
			{
				bytes[i] = 0;
			}
		}
	}
	
	public RHP_Timeout()
	{
		m_Name = "RHP_Timeout";
	}
	
}

}

File: examples/IOP-Dynamic-ID-Assignment/Client_2/src/urn_jts_DiscoveryClient_1_0/InternalEvents/RHP_Timeout.cpp
#include "urn_jts_DiscoveryClient_1_0/InternalEvents/RHP_Timeout.h"

namespace urn_jts_DiscoveryClient_1_0
{

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int RHP_Timeout::getSize()
{
	unsigned int size = 0;
	
	
	return size;
}

void RHP_Timeout::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
}

void RHP_Timeout::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
}

RHP_Timeout::RHP_Timeout()
{
	m_Name = "RHP_Timeout";
}

RHP_Timeout::~RHP_Timeout()
{
}


}
